java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowConfig.java

```diff
@@ -53,6 +53,21 @@
   private Map<Integer, JdbcFieldInfo> arraySubTypesByColumnIndex;
   private Map<String, JdbcFieldInfo> arraySubTypesByColumnName;
 
+  public static final int DEFAULT_TARGET_BATCH_SIZE = 1024;
+  public static final int NO_LIMIT_BATCH_SIZE = -1;
+
+  /**
+   * The maximum rowCount to read each time when partially convert data.
+   * Default value is 1024 and -1 means disable partial read.
+   * default is -1 which means disable partial read.
+   * Note that this flag only useful for {@link JdbcToArrow#sqlToArrowVectorIterator}
+   * 1) if targetBatchSize != -1, it will convert full data into multiple vectors
+   * with valueCount no more than targetBatchSize.
+   * 2) if targetBatchSize == -1, it will convert full data into a single vector in {@link ArrowVectorIterator}
+   * </p>
+   */
```

Comment:
this is probably more useful, on the getter/setter instead of the private member?

Reply:
This value could specify use JdbcToArrowConfig, this default value is useful only when users not set this value.